### PR TITLE
Move original version to cold storage after successful update

### PR DESCRIPTION
**Pull request by bcarthic**
## Description
Move original version to cold storage after successful update

## Related issues
Addresses [[AB#103334](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/103334)].

## Testing
Added new unit tests

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs [hunk 1]
```diff
@@ -128,4 +128,14 @@ public interface IFileStore
...
     Task CopyFileAsync(long originalVersion, long newVersion, Partition partition, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously change blob access tier to cold tier.
+    /// </summary>
+    /// <param name="version">The DICOM instance version.</param>
+    /// <param name="partition">Partition to use when operating on file</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>Key value pair of blockId and blockLength.</returns>
+
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs:139]: nit: extra newline

File: src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs [hunk 2]
```diff
@@ -128,4 +128,14 @@ public interface IFileStore
...
     /// <returns></returns>
     Task CopyFileAsync(long originalVersion, long newVersion, Partition partition, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously change blob access tier to cold tier.
+    /// </summary>
+    /// <param name="version">The DICOM instance version.</param>
+    /// <param name="partition">Partition to use when operating on file</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>Key value pair of blockId and blockLength.</returns>
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs:138]: Only returns a task 😛

File: src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs [hunk 3]
```diff
@@ -230,6 +234,19 @@ public partial class UpdateDurableFunction
...
                 }
+
+                if (!isFailedToUpdateStudy)
+                {
+                    await context.CallActivityWithRetryAsync(
+                     nameof(DeleteOldVersionBlobV2Async),
+                     _options.RetryOptions,
+                     new CleanupBlobArguments(instanceWatermarks, input.Partition));
+
+                    await context.CallActivityWithRetryAsync(
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs:245]: This is probably not actually a breaking change as the activities are at the end... I think lol. Although is `DeleteOldVersionBlobV2Async` executing now in a different circumstance?

File: src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs [hunk 4]
```diff
@@ -230,6 +234,19 @@ public partial class UpdateDurableFunction
                     // Cleanup the new version when the update activity fails
                     await TryCleanupActivityV2(context, instanceWatermarks, input.Partition);
                 }
+
+                if (!isFailedToUpdateStudy)
+                {
+                    await context.CallActivityWithRetryAsync(
+                     nameof(DeleteOldVersionBlobV2Async),
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs:241]: nit: formatting is messed up

File: src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs [hunk 5]
```diff
@@ -393,5 +393,46 @@ public async Task CleanupNewVersionBlobV2Async([ActivityTrigger] CleanupBlobArgu
         logger.LogInformation("New blobs deleted successfully. Total size {TotalCount}", fileCount);
     }
 
+    /// <summary>
+    /// Asynchronously move all the original version blobs to cold access tier.
+    /// </summary>
+    /// <param name="arguments">arguments which have a list of watermarks to move to cold access tier along with partition they belong to</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>
+    /// A task representing the <see cref="CleanupNewVersionBlobV2Async"/> operation.
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs:402]:
```suggestion
    /// A task representing the <see cref="SetOriginalBlobToColdAccessTierAsync"/> operation.
```

File: src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs (hunk 3 above)
Comment (bcarthic) [reply src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs:245]: Yeah, I found a bug. Not sure how it missed. DeleteOldVersionBlobV2Async was only called for last studyInstanceUId. But it should be called for every commit.

Comment (bcarthic) [reply src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs:245]: Also we don't have delete the old version blobs if the current activity run returns an error.

**Review by wsugarman — APPROVED** [review]

**Pull request merged**